### PR TITLE
Reformatted codebase to follow pep8 guidelines

### LINKED ISSUE #101: Reformat the code
Re-Formatting the code to follow the pip8 style guide i.e. No trailing whitespaces, No more than 79 characters per line etc.

### PR DESCRIPTION
**Pull request by Monal5031**
Used coala to fix all formatting issues locally.
Integrated coala with travis to raise formatting issues in PR itself from now on.

Fixes #101

### CODE REVIEW

**Comment by Monal5031** [comment]
@pombredanne @singh1114 @RajuKoushik Please review this PR the travis is now green 😄 , sorry for late fix didn't get a chance to update the branch.
I've used `coala` locally to fix the formatting issues and if this is merged the formatting errors will be shown in PR itself under the `python 3.6` tab of travis. Also coala follows `pep8` guidelines while formatting.

File: scanapp/migrations/0001_initial.py
Comment (pombredanne) [inline scanapp/migrations/0001_initial.py]: Please do NOT reformat or change auto-generated migrations

File: .travis.yml
Comment (pombredanne) [inline .travis.yml]: We do not support Python 3.6 yet at this stage since ScanCode does not support it for now

File: .travis.yml
Comment (pombredanne) [inline .travis.yml]: I am not sure I understand why 3.6 is needed and what it does?

File: .travis.yml
Comment (pombredanne) [inline .travis.yml]: Same as above: I am confused by the tests on the python version.

File: scanapp/migrations/0002_auto_20170815_2324.py
Comment (pombredanne) [inline scanapp/migrations/0002_auto_20170815_2324.py]: Same: do not reformat migrations

File: scanapp/models.py
Comment (pombredanne) [inline scanapp/models.py:45]: What is this extra line about?

File: scanapp/models.py
Comment (pombredanne) [inline scanapp/models.py]: I am not too keen on this type of strict PEP8 line wrapping for the sake of it. I consider that lines should be ~ 80ish in length. Here you are reformatting readable code in something that is no longer readable. What's the point?

**Review by pombredanne — CHANGES_REQUESTED** [review]
Thank you!... See my comments: most of your proposed changes are about strict PEP8 line wrapping for the sake of it. I consider that lines should be ~ 80ish in length and line wrapping for the sake of PEP8 in many case destroy readability. Could you give a shot at viewing https://www.youtube.com/watch?v=wf-BqAjZb8M

File: scanapp/models.py
Comment (Monal5031) [reply scanapp/models.py:45]: `Method definitions inside a class are surrounded by a single blank line.`
This is a point in pep8 guideline, I guess that's why coala added a new line while re-formatting.

File: scanapp/migrations/0001_initial.py
Comment (Monal5031) [reply scanapp/migrations/0001_initial.py]: Sure, let me try something to ignore migrations directory.

File: .travis.yml
Comment (Monal5031) [reply .travis.yml]: The thing is scancode works on python2.7 while coala works on python3+ versions. So to cater to both of them I added this workaround, which will run travis on 2 different versions of python, 2.7  for scancode and 3.6 for coala.

File: scanapp/models.py
Comment (Monal5031) [reply scanapp/models.py]: @pombredanne If you wish to increase the number of characters allowed per line, coala has a feature for that too. We can define ` max_line_length ` according to our need. 😄

File: .travis.yml
Comment (Monal5031) [reply .travis.yml]: Yeah keeping that in mind I added these two versions, as coala will need python3+ versions. You can see in this file that I used conditional logic to run ScanCode in python2.7 and coala in python3.6

Comment (pombredanne) [reply .travis.yml]: Cannot you just run coala in the same build run? e.g. python3 is likely available in both cases.

File: scanapp/models.py
Comment (pombredanne) [reply scanapp/models.py]: I did not see any of the proposed line breakings that made the code more beautiful and easier to read.

File: .travis.yml
Comment (Monal5031) [reply .travis.yml]: I tried running it in same build but travis failed each time, here are the failed logs:
https://travis-ci.org/nexB/scancode-server/builds/288168067
https://travis-ci.org/nexB/scancode-server/builds/288178128
https://travis-ci.org/nexB/scancode-server/builds/289554875
I'll give it a try running it in the same build, if you wish.

File: scanapp/models.py
Comment (Monal5031) [reply scanapp/models.py]: Okay if it isn't much of a use, then the better alternative I guess is to drop the idea of reformatting. your views?

File: scanapp/migrations/0001_initial.py
Comment (jayvdb) [reply scanapp/migrations/0001_initial.py]: Or improve the generator?

File: .travis.yml
Comment (jayvdb) [reply .travis.yml]: It is usually better to run the linters in a separate job, so that they run concurrently with the unit tests

You can use `matrix: include: ` to add an extra job for `coala-ci`.  The matrix entry can have its own `script:` etc, so you do not need to modify the main job.

**Review by jayvdb — CHANGES_REQUESTED** [review]

File: scanapp/migrations/0001_initial.py
Comment (pombredanne) [reply scanapp/migrations/0001_initial.py]: @jayvdb this is the generator that is built-in Django and I am not sure it makes sense to spend any time to update this. Generated code should never be linted IMHO.

File: scanapp/models.py
Comment (pombredanne) [reply scanapp/models.py]: Formatting is fine, as long as we have the right rules such as line length in the 80ish range. Skip migrations, etc...

File: scanapp/models.py
Comment (jayvdb) [reply scanapp/models.py:45]: Ya, there should be a blank line above methods, like `__str__`.
This rule can be disabled if scancode maintainers prefer to not have a space here.

File: scanapp/models.py
Comment (jayvdb) [reply scanapp/models.py]: It looks like this repo should have `max_line_length = 100`.  There should only be a few instances of violations of that limit.
Then it will be easier to review and consider those violations.

If there are still more than 20 violations of `max_line_length = 100` , increase it to `max_line_length = 120`.
The goal of adding a `.coafile` is _not_ to fix all the problems, but instead it is to prevent future regressions making the codebase style even less unified.
After the `.coafile` is merged and being checked by CI, additional improvements can be made to style rules incrementally.

**Review by jayvdb — CHANGES_REQUESTED** [review]

File: scanapp/models.py
Comment (Monal5031) [reply scanapp/models.py]: Okay will do it asap :+1:

**Comment by Monal5031** [comment]
@pombredanne @jayvdb Please take a look now. I tried `max_line_length=100` but it still had many violations, so updated with `120`.

**Comment by pombredanne** [comment]
@Monal5031 this looks decent now alright .... but since this is touching several files, you now have a responsibility I merge this now: You will need to rebase and/or resolve any conflicts in the pending PRs in https://github.com/nexB/scancode-server/pulls ? Do you think you could do this? that should not be a huge amount of work. I can even bring in these PR  branches in this repo if this helps.

**Comment by Monal5031** [comment]
@pombredanne Okay sure, I checked. this change won't cause much merge conflicts, though we'll be needing rebases over all PRs. Also, some of them might fail Travis as they will also need to run coala. I can fix this repository if it breaks 😄

**Comment by pombredanne** [comment]
@Monal5031 I take your word for it that you fix the mess that may come! So please start working on it now :P

**Comment by Monal5031** [comment]
Okay @pombredanne will head onto them as soon as this is merged 😃

**Comment by pombredanne** [comment]
You will also need to adjust your commit comment style to be imperative in the future.

**Comment by Monal5031** [comment]
Sure 👍 , working on the old PRs

**Pull request merged**
